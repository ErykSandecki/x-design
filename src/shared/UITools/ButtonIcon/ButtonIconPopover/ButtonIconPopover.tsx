import { FC, RefObject } from 'react';

// components
import Box from '../../../UI/Box/Box';
import Popover, { PopoverCompound, TPopoverProps } from '../../Popover/Popover';

export type TButtonIconPopoverProps = Pick<TPopoverProps, 'children' | 'offset'> & {
  ref: RefObject<HTMLDivElement>;
  selected: boolean;
  setSelected: TFunc<[boolean]>;
};

export const ButtonIconPopover: FC<TButtonIconPopoverProps> = ({ children, offset, ref, selected, setSelected }) => (
  <Box ref={ref} sx={{ alignItems: 'center', display: 'flex', justifyContent: 'center', position: 'relative' }}>
    <Popover backgroundColor="neutral5" e2eValue="popover" offset={offset} refItem={ref} selected={selected}>
      <PopoverCompound.PopoverRoot setSelected={setSelected}>{children}</PopoverCompound.PopoverRoot>
    </Popover>
  </Box>
);

export default ButtonIconPopover;
