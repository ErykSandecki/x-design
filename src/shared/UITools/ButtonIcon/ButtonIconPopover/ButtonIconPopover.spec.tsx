import { RefObject } from 'react';
import { noop } from 'lodash';

// components
import ButtonIconPopover from './ButtonIconPopover';

// utils
import { customRender } from 'test';

const ref = { current: {} } as RefObject<HTMLDivElement>;

describe('TextFieldPopover snapshots', () => {
  it('should render ButtonIconPopover', () => {
    // before
    const { asFragment } = customRender(
      <ButtonIconPopover ref={ref} selected={false} setSelected={noop}>
        children
      </ButtonIconPopover>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with opened popover', () => {
    // before
    const { asFragment } = customRender(
      <ButtonIconPopover ref={ref} selected setSelected={noop}>
        children
      </ButtonIconPopover>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
