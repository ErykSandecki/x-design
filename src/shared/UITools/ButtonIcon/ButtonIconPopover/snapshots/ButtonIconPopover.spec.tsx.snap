// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldPopover snapshots should render ButtonIconPopover 1`] = `
<DocumentFragment>
  <div
    class="css-zcp5dz"
    data-test-box=""
  >
    <div
      class="css-6wln18 Popover Popover--light"
      data-test-box=""
      data-test-popover="popover"
      style="left: 0px; top: 0px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextFieldPopover snapshots should render with opened popover 1`] = `
<DocumentFragment>
  <div
    class="css-zcp5dz"
    data-test-box=""
  >
    <div
      class="css-6wln18 Popover Popover--light Popover--selected Popover--selected--light"
      data-test-box=""
      data-test-popover="popover"
      style="left: 0px; top: 0px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;
